Municipality/azc.mod2g

```diff
@@ -1,13 +1,32 @@
-use dummy as knowledge.
-
-order = linearall.
-module azc{
-	%Check if there is still need for an azc (id 650)
-	if bel(building(BuildingID,_,OwnerID,_,_,650,_)) then exit-module.
-	%Check if we have budget for an azc
-	if bel(money(MoneyAmount),MoneyAmount<100000) then exit-module.
-	%Check if there is space for an azc
-
-	%Adopt if all preconditions are met
-	if true then adopt(buildAZC).
-}
\ No newline at end of file
+use dummy as knowledge.
+
+order = linearall.
+module azc{
+	%if we have build an azc in the last ten cycles wait for indicators to update
+	if bel(azc_timeout(X), X>0, Y is X-1) then
+		delete(azc_timeout(X)) + insert(azc_timeout(Y)).
+	if bel(azc_timeout(X), X>0) then
+		exit-module.

```

Comment:
Creative solution, I think this will suffice for accounting for non direct interaction between the agent actions and the environment changing.